test/integration/serverTest.js

```diff
@@ -976,4 +1009,85 @@ describe('Brave Sensor server', () => {
+      await db.clearClients()
+
+      const client = await clientFactory(db)
+      await db.createLocation(
+        testLocation1Id,
+        testLocation1PhoneNumber,
+        MOVEMENT_THRESHOLD,
+        STILLNESS_TIMER,
+        DURATION_TIMER,
+        1000,
+        INITIAL_TIMER,
+        ['+15005550006'],
+        '+15005550006',
+        ['+15005550006'],
+        1000,
+        'locationName',
+        radar_coreID,
+        radar_coreID,
+        'Innosent',
+        'alertApiKey',
+        true,
+        true,
+        null,
+        firstLowBatteryAlert,
+        client.id,
+      )
+      sandbox.stub(braveAlerter, 'startAlertSession')
+      sandbox.stub(braveAlerter, 'sendSingleAlert')
+    })
+
+    afterEach(async () => {
+      await redis.clearKeys()
+      await db.clearSessions()
+      await db.clearLocations()
+      await db.clearClients()
+      sandbox.restore()
+      helpers.log('\n')
+    })
+
+    it('should return 200 for a heartbeat request with no headers', async () => {
+      const response = await chai.request(server).post('/api/heartbeat').send({})
+      expect(response).to.have.status(200)
+    })
+
+    it('should return 200 for a valid heartbeat request', async () => {
+      const request = {
+        coreid: radar_coreID,
+        data: `{"doorMissedMsg": 0, "doorLowBatt": true, "doorLastHeartbeat": 1000, "resetReason": "NONE", "states":[]}`,
+      }
+      const response = await chai.request(server).post('/api/heartbeat').send(request)
+      expect(response).to.have.status(200)
+    })
+
+    it('should not update sentLowBatteryAlertAt if battery is normal', async () => {
+      const testLocation = await db.getLocationData(testLocation1Id)
+      await normalHeartbeat(radar_coreID)
+      const locationAfterHeartbeat = await db.getLocationData(testLocation1Id)
+      expect(locationAfterHeartbeat.sentLowBatteryAlertAt).to.deep.equal(testLocation.sentLowBatteryAlertAt)
+    })
+
+    it('should update sentLowBatteryAlertAt if the battery is low and the period since the last alert is greater than the timeout', async () => {
+      const locationBeforeHeartbeat = await db.getLocationData(testLocation1Id)
+      await lowBatteryHeartbeat(radar_coreID)
+      const locationAfterHeartbeat = await db.getLocationData(testLocation1Id)
+      expect(locationBeforeHeartbeat.sentLowBatteryAlertAt).to.not.deep.equal(locationAfterHeartbeat.sentLowBatteryAlertAt)
+    })
+
+    it('should not update sentLowBatteryAlertAt if the battery is low and the period since the last alert is less than the timeout', async () => {
+      await lowBatteryHeartbeat(radar_coreID)
+      const locationAfterHeartbeat = await db.getLocationData(testLocation1Id)
+      await lowBatteryHeartbeat(radar_coreID)
+      const locationAfterTwoHeartbeats = await db.getLocationData(testLocation1Id)
+      expect(locationAfterTwoHeartbeats.sentLowBatteryAlertAt).to.deep.equal(locationAfterHeartbeat.sentLowBatteryAlertAt)
```

Comment:
Could you please add. tests to show the the `sendSingleAlert` is called or not in each of these cases? And whether `logSentry`  is called or not in each of these cases?